File: front-end/lib/model/issue.dart

import 'package:json_annotation/json_annotation.dart';
import 'package:scrum_master_front_end/model/project.dart';
import 'package:scrum_master_front_end/model/sprint.dart';
import 'package:scrum_master_front_end/model/user.dart';

part 'issue.g.dart';

@JsonSerializable(explicitToJson: true)
class Issue {
  int? id;
  String? code;
  String? title;
  String? description;
  int? estimate;
  String? label;
  String? type;
  User? reporter;
  User? assignee;
  Sprint? sprint;
  int? priority;
  Project? project;
  List<Issue>? subTasks;
  String? status;


  Issue(
      this.id,
      this.code,
      this.title,
      this.description,
      this.estimate,
      this.label,
      this.type,
      this.reporter,
      this.assignee,
      this.sprint,
      this.priority,
      this.project,
      this.status,
      this.subTasks);

  factory Issue.fromJson(Map<String, dynamic> json) => _$IssueFromJson(json);

  Map<String, dynamic> toJson() => _$IssueToJson(this);
}

File: back-end/src/main/java/com/scrum/master/data/entities/Issue.java
package com.scrum.master.data.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.scrum.master.common.enums.IssueStatus;
import com.scrum.master.common.enums.IssueType;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "issues")
@Getter
@Setter
public class Issue extends BaseEntity {

    private int priority;
    private String code;
    private String title;
    private String description;
    private String label;
    @Enumerated(EnumType.STRING)
    private IssueType type;
    private Integer estimate;
    @Enumerated(EnumType.STRING)
    private IssueStatus status = IssueStatus.Todo;


    @OneToMany
    @JoinColumn(name = "issue_id")
    private List<Issue> subTasks;

    @ManyToOne
    @JoinColumn(name = "reporter")
    private User reporter;

    @ManyToOne
    @JoinColumn(name = "assignee")
    private User assignee;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "sprint_id")
    @JsonIgnore
    private Sprint sprint;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "project_id")
    private Project project;

    @Transient
    private Long sprintId;

}
